src/SwedbankPay/Api/Client/Version.php

```diff
@@ -48,14 +50,91 @@ protected function getVersionFromEnvironment()
             return getenv("VERSION");
         }
 
-        $path = __DIR__ . '/../../../../composer.json';
-        $contents = file_get_contents($path);
-        $data = json_decode($contents, true);
+        $composerPath = __DIR__ . '/../../../../composer.json';
+        $composer = $this->readJson($composerPath);
+
+        if (isset($composer['version'])) {
+            return $composer['version'];
+        }
 
-        if (isset($data['version'])) {
-            return $data['version'];
+        $composerLockPath = getcwd() . DIRECTORY_SEPARATOR . 'composer.lock';
+        $composerLock = $this->readJson($composerLockPath);
+
+        if (isset($composerLock['packages'])) {
+            $packages = $composerLock['packages'];
+            foreach ($packages as $package) {
+                if (!isset($package['name']) ||
+                    $package['name'] != "swedbank-pay/swedbank-pay-sdk-php") {
+                    continue;
+                }
+
+                if (isset($package['version'])) {
+                    return $package['version'];
+                }
+            }
         }
 
         throw new ClientException('VERSION not found in environment variable, composer.json or anywhere else.');
     }
+
+
+    /**
+     * Reads the file at $path and returns it as a JSON decoded object.
+     *
+     * @param $path The path of the file to read and JSON decode.
+     * @return object The JSON decoded object.
+     */
+    private function readJson($path)
+    {
+        $path = $this->truePath($path);
+        $contents = file_get_contents($path);
+        return json_decode($contents, true);
+    }
+
+
+    /**
+    * This function is to replace PHP's extremely buggy realpath().
+    * https://stackoverflow.com/a/4050444/61818
+    *
+    * @param string The original path, can be relative etc.
+    * @return string The resolved path, it might not exist.
+    */
+    private function truePath($path){
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method truePath() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5303907)